src/code/z_actor.c

```diff
@@ -2807,17 +2807,19 @@ Actor* Actor_Spawn(ActorContext* actorCtx, PlayState* play, s16 actorId, f32 pos
             osSyncPrintf(VT_FGCOL(GREEN));
             osSyncPrintf("OVL(a):Seg:%08x-%08x Ram:%08x-%08x Off:%08x %s\n", overlayEntry->vramStart,
                          overlayEntry->vramEnd, overlayEntry->loadedRamAddr,
-                         (u32)overlayEntry->loadedRamAddr + (u32)overlayEntry->vramEnd - (u32)overlayEntry->vramStart,
-                         (u32)overlayEntry->vramStart - (u32)overlayEntry->loadedRamAddr, name);
+                         (uintptr_t)overlayEntry->loadedRamAddr + (uintptr_t)overlayEntry->vramEnd -
+                             (uintptr_t)overlayEntry->vramStart,
+                         (uintptr_t)overlayEntry->vramStart - (uintptr_t)overlayEntry->loadedRamAddr, name);
             osSyncPrintf(VT_RST);
 
             overlayEntry->numLoaded = 0;
         }
 
-        actorInit = (void*)(u32)((overlayEntry->initInfo != NULL)
-                                     ? (void*)((u32)overlayEntry->initInfo -
-                                               (s32)((u32)overlayEntry->vramStart - (u32)overlayEntry->loadedRamAddr))
-                                     : NULL);
+        actorInit = (void*)(uintptr_t)(
+            (overlayEntry->initInfo != NULL)
+                ? (void*)((uintptr_t)overlayEntry->initInfo -
+                          (intptr_t)((uintptr_t)overlayEntry->vramStart - (uintptr_t)overlayEntry->loadedRamAddr))
```

Comment:
Casting a uintptr_t-uintptr_t to intptr_t here, shouldn't that be the inexistent "signed size_t" or ptrdiff_t instead?

Reply:
Maybe but that would mean figuring out which type to use and I'd rather not get into that debate here😅(we don't have either of these types in the repo rn btw)

At least for now intptr_t works fine imo and should make it 64 bit compatible

Reply:
compared to a s32 cast it would be harder to catch that for fixing up later

Reply:
A s32 cast wouldn't be 64 bit compatible here so I would rather not downgrade it. I think it would be better to keep a note that this pattern could be revisited if we decide to use some form of signed size_t

Reply:
This is where I'm not sure what the goal behind using the proper types is, I would assume proper C but using `(intptr_t)` here is not proper C (it's not specified as able to hold a difference between `uintptr_t`s), nor is `(s32)` of course.
I hear the argument that it makes it 64bit compatible but that's just an implementation detail of the 64bit implementations we know of, honestly as a temporary fix I would rather see a new type introduced now, perhaps only temporarily, than introducing a temporary mistake
We could even do something stupid like
```suggestion
#ifndef NON_MATCHING
                          (s32)
#endif
                          ((uintptr_t)overlayEntry->vramStart - (uintptr_t)overlayEntry->loadedRamAddr))
```

Reply:
As far as I'm concerned, the main goal behind using intptr types is to to make the code 64 bit compatible (otherwise I think it's not worth it if the types are just meant as an alias to s32/u32). I agree `intptr_t` isn't the most ideal type here, but afaik `intptr_t` is really just defined as an integer type capable of holding a pointer which is all we need here, so I think it's "fine" to use it in this context.

Also fwiw `intptr_t` is what MM has been using in this pattern, and it was part of what we wanted to merge in [1334](https://github.com/zeldaret/oot/pull/1334) (without the macros)

Reply:
I don't really mind reverting it but I think it would be unfortunate and it's definitely going to come up again in the future